cardinal/message/manager.go

```diff
@@ -1,27 +1,31 @@
 package message
 
 import (
+	"reflect"
+	"slices"
+
 	"github.com/rotisserie/eris"
 	"pkg.world.dev/world-engine/cardinal/types"
-	"slices"
 )
 
 type Manager struct {
-	registeredMessages map[string]types.Message
-	nextMessageID      types.MessageID
+	registeredMessagesByName map[string]types.Message
+	registeredMessagesByType map[reflect.Type]types.Message
+	nextMessageID            types.MessageID
 }
 
 func NewManager() *Manager {
 	return &Manager{
-		registeredMessages: map[string]types.Message{},
-		nextMessageID:      1,
+		registeredMessagesByType: map[reflect.Type]types.Message{},
+		registeredMessagesByName: map[string]types.Message{},
+		nextMessageID:            1,
 	}
 }
 
-// RegisterMessages registers multiple messages with the message manager
+// RegisterMessagesByName registers multiple messages with the message manager
 // There can only be one message with a given name, which is declared by the user by implementing the Name() method.
 // If there is a duplicate message name, an error will be returned and none of the messages will be registered.
-func (m *Manager) RegisterMessages(msgs ...types.Message) error {
+func (m *Manager) RegisterMessagesByName(msgs ...types.Message) error {
```

Comment:
To be consistent with the rest of the manager, this should be `RegisterMessages`

Reply:
Sure I don't mind, I was just renaming it to keep it consistent witht he lower level method call.